dist/spec/verifiable-payid-protocol-NC.txt

```diff
@@ -0,0 +1,896 @@
+
+3.1.2.  JWS Payload
+
+   The JWS payload is the message that needs to be signed.
+
+
+
+
+
+
+
+
+
+Malhotra & Schwartz     Expires January 31, 2021                [Page 6]
+
+Internet-Draft     Verifiable PayID Protocol Extension         July 2020
+
+
+  {
+          "exp"  : "",
+          "payId": "bob$wallet.com",
+          "payIdAddress": {
+            "expTime": "",
+            "paymentNetwork": "XRPL",
+            "environment": "TESTNET",
+            "addressDetailsType": "CryptoAddressDetails",
+            "addressDetails": {
+              "address": "T7CKYKhRujaxEs9fSxQwJApHsQVPKUgD7EtLWCGTAFBwTha"
+            }
+          }
+  }
+
+3.1.2.1.  exp
+
+   The "exp" field is an optional field as described in [RFC7797].  If
+   used, it SHOULD be set to the expiration time of the cryptographic
+   key used to generate the digital signature.
+
+3.1.2.2.  payId
+
+   The "payId" field is a required field.  The value of "payId" field is
+   the PayID URI in the client request that identifies the payment
+   account information that the JSON object describes.
+
+3.1.2.3.  PayIDAddress
+
+   The "PayIDAddress" is a required field.  The value of "PayIDAddress"
+   field is a JSON object with the following keys:
+
+   o  "expTime": This is an optionl field.  The value of "expTime"
+      string is set to the maximum time upto which the payment address
+      in the "address" field is valid.
+
+   o  "paymentNetwork": The value of the "paymentNetwork" is the value
+      of payment-network string as specified in the client request's
+      "Accept" header.
+
+   o  "environment": The value of "environment" string is the value of
+      environment as specified in the client request's "Accept" header.
+
+   o  "addressDetailsType": The value of "addressDetailsType" is one of
+      the following strings:
+
+      *  CryptoAddressDetails
+
+      *  FiatAddressDetails
+
+
+
+Malhotra & Schwartz     Expires January 31, 2021                [Page 7]
+
+Internet-Draft     Verifiable PayID Protocol Extension         July 2020
+
+
+   o  "addressDetails": The value of "addressDetails" is the address
+      information necessary to send payment on a specific
+      "paymentNetwork" and "environment".
+
+   The "address" field MUST be present in the JWS payload.
+
+3.1.3.  JWS signature
+
+   The JWS signature is the digital signature which is calculated over
+   the JOSE header and the JWS payload.
+
+    "signature": "base64Signature"
+
+3.1.3.1.  signature
+
+   The value of "signature" field is computed as described in [RFC7515].
+
+3.2.  End-to-End Verifiable PayID Protocol Flow for Non-Custodial
+      Wallets and Exchanges
+
+   A pre-requisite for this extension of Verifiable PayID protocol
+   requires the PayID owner to transfer signed "PaymentInformation" to
+   the PayID server.  In this document we specify one such way of doing
+   this.
+
+   Following are the pre-steps at PayID owner's (non-custodial wallet's
```

Comment:
Consider `The following are the pre-steps that a PayID owner's device should perform locally:`